### DISCUSSION TITLE
when histogram observe over (1 >> 32 - 2) times

### DISCUSSION THREAD

**Discussion by greymonster-mm** [prometheus/client_golang#1115] [General]
when histogram observe over (1 >> 32 - 2) times , the write method will block....

<img width="778" alt="image" src="https://user-images.githubusercontent.com/18045224/184105355-5424dd54-6751-4a63-9054-9ef6dad59ebb.png">

countAndHotIdx use lower 63 bit as count , but hotCounts.count use 64 bit , when histogram observe over (1 >> 32 - 2) times , here will block always....

<img width="869" alt="image" src="https://user-images.githubusercontent.com/18045224/184105601-58722cc1-a09a-41d7-9aee-4eb8a15c60a2.png">

**Comment by beorn7** [comment]
I would be surprised if it happened in that way, but perhaps I'm misunderstanding you. Do you have a code example to reproduce the behavior? That would clarify a lot and would help to debug.
